packages/next-swc/crates/next-core/src/next_font/local/font_fallback.rs

```diff
@@ -52,3 +74,369 @@ pub(super) async fn get_font_fallbacks(
 
     Ok(FontFallbacksVc::cell(font_fallbacks))
 }
+
+async fn get_font_adjustment(
+    context: FileSystemPathVc,
+    options: NextFontLocalOptionsVc,
+    fallback_font: &DefaultFallbackFont,
+) -> Result<FontAdjustment> {
+    let options = &*options.await?;
+    let main_descriptor = pick_font_for_fallback_generation(&options.fonts)?;
+    let font_file = &*context.join(&main_descriptor.path).read().await?;
+    let font_file_rope = match font_file {
+        FileContent::NotFound => bail!("Expected font file content"),
+        FileContent::Content(file) => file.content(),
+    };
+
+    let mut font_file_binary = Vec::with_capacity(font_file_rope.len());
+    for bytes in font_file_rope.read() {
+        font_file_binary.extend(bytes);
+    }
```

Comment:
For file reads, we're actually guaranteed (currently) to have a single backing bytes buffer in the rope. We could optimize this by adding a new `to_bytes() -> Cow<'a, [u8]>` method to Rope that does the same thing as `to_str()`.